File: c/lc/triangle.js
/**
 * @param {number[][]} triangle
 * @return {number}
 */
let minimumTotal = function(triangle) {
    let sz = triangle.length;
    let dp = triangle[sz-1];
    for (r=sz-2; r>=0; r--)
        for (c=0; c<=r; c++)
            dp[c] = triangle[r][c] + Math.min(dp[c], dp[c+1]);
    return dp[0];
};

File: c/lc/triangle.cc
// https://leetcode.com/problems/triangle/

class Solution {
public:
    int minimumTotal(vector<vector<int>>& tri) {
        int h = tri.size();
        for (int r = h-2; r >= 0; r--)
            for (int c = r; c >= 0; c--)
                tri[r][c] += min(tri[r+1][c], tri[r+1][c+1]);
        return tri[0][0];
    }
};
